project/canvas.py

```diff
@@ -0,0 +1,650 @@
+"""
+This module contains the canvas and entry section items that fill
+the normal tk window (located in __main__.py)
+"""
+
+from io import BytesIO
+import typing
+import pathlib
+import random
+
+import aiofiles
+import asynctk as tk
+from tkinter.ttk import Progressbar
+from PIL import Image, ImageDraw, ImageTk
+import asyncio
+
+
+async def start(bar):
+    while 1:
+        bar.step()
+        await asyncio.sleep(0.05)
+
+
+class Colour:
+
+    """
+    The colour class - used to unify all representations of colour as needed
+    by third-party modules.
+
+    This class also switches the colour around to fit the theme of the code jam.
+
+
+    Parameters
+    ----------
+    colour: int or str
+        The colour inputted (given by the text box Entry)
+
+    All examples are with the Colour initialised with Colour("15715755")
+
+    Attributes
+    ----------
+    fake_colour: str
+        The colour in hex before reformatting
+        e.g. "efcdab"
+    r: str
+        The amount of red in hex format.
+        e.g. "ab"
+    g: str
+        The amount of green in hex format.
+        e.g. "cd"
+    b: str
+        The amount of blue in hex format.
+        e.g. "ef"
+    colour: str
+        The colour in hex after the format is switched.
+        e.g. "abcdef"
+    as_hex: str
+        The colour prefixed with #
+        This is the most common way to represent a colour, and the main one
+        used by TK/TCL.
+        e.g. "#abcdef"
+    as_int: int
+        The colour in an integer with the hex converted into denary.
+        e.g. 11259375
+    as_rgb: tuple[int]
+        The colour in an (r, g, b) tuple.
+        e.g. (171, 205, 239)
+
+    Methods
+    -------
+    from_rgb: classmethod
+        Creates class from an (r, g, b) tuple.
+
+    """
+
+    def __init__(self, colour: typing.Union[str, int]):
+        try:
+            int(colour)
+        except ValueError:
+            raise TypeError
```

Comment:
this is a bit of an antipattern. you should raise this error with a message explaining what's happening, or it becomes very frustrating to debug.

Reply:
I'll leave this one to @Suhail6inkling , who wrote this code.